src/mod_adlib.erl

```diff
@@ -25,10 +25,291 @@
+        [ Endpoint#adlib_endpoint.name, Since ],
+        Context),
+    import_all_loop(m_adlib_api:fetch_since(Endpoint, Since, Context), 0, Endpoint, Context).
+
+
+import_all_loop(done, Count, _Endpoint, _Context) ->
+    {ok, Count};
+import_all_loop({error, _} = Error, _Count, _Endpoint, _Context) ->
+    Error;
+import_all_loop({ok, {Recs, Cont}}, Count, Endpoint, Context) ->
+    Count1 = lists:foldl(
+        fun(#{ <<"@attributes">> := #{ <<"priref">> := PrirefAttr } }, Acc) ->
+            Priref = z_convert:to_integer(PrirefAttr),
+            case import_record(Endpoint, Priref, Context) of
+                {ok, Id} ->
+                    ?zInfo(
+                        "Adlib: import from endpoint '~s' priref '~p' to rsc id ~p",
+                        [ Endpoint#adlib_endpoint.name, Priref, Id ],
+                        Context),
+                    Acc+1;
+                {error, Reason} ->
+                    ?zWarning(
+                        "Adlib: import from endpoint '~s' priref '~p' error ~p",
+                        [ Endpoint#adlib_endpoint.name, Priref, Reason ],
+                        Context),
+                    Acc
+            end
+        end,
+        Count,
+        Recs),
+    case Cont of
+        {next, ContFun} ->
+            ?MODULE:import_all_loop(ContFun(), Count1, Endpoint, Context);
+        done ->
+            ?zInfo(
+                "Adlib: import from endpoint '~s' finished, imported ~p documents",
+                [ Endpoint#adlib_endpoint.name, Count1 ],
+                Context),
+            {ok, Count1}
+    end.
+
+
+%% @doc Fetch a specific record from an endpoint, the Adlib JSON is returned.
+-spec fetch_record( Endpoint, Priref, Context ) -> {ok, AdlibJSON} | {error, term()}
+    when Endpoint :: mod_adlib:adlib_endpoint() | binary(),
+         Priref :: priref(),
+         Context :: z:context(),
+         AdlibJSON :: map().
+fetch_record(Name, Priref, Context) when is_binary(Name) ->
+    case name_to_endpoint(Name, Context) of
+        {ok, E} -> fetch_record(E, Priref, Context);
+        {error, _} = Error -> Error
+    end;
+fetch_record(#adlib_endpoint{} = Endpoint, Priref, Context) ->
+    m_adlib_api:fetch_record(Endpoint, Priref, Context).
+
+
+%% @doc Import a specific record from an endpoint.
+-spec import_record( Endpoint, Priref, Context ) -> {ok, RscId} | {error, term()}
+    when Endpoint :: mod_adlib:adlib_endpoint() | binary(),
+         Priref :: priref(),
+         Context :: z:context(),
+         RscId :: m_rsc:resource_id().
+import_record(Name, Priref, Context) when is_binary(Name) ->
+    case name_to_endpoint(Name, Context) of
+        {ok, E} -> import_record(E, Priref, Context);
+        {error, _} = Error -> Error
+    end;
+import_record(#adlib_endpoint{} = Endpoint, Priref, Context) ->
+    case triples(Endpoint, Priref, Context) of
+        {ok, Triples} ->
+            Options = [
+                {import_edges, 1},
```

Comment:
Does `import_edges, 1` mean that nodes at outgoing edges are fetched, one level deep?

Reply:
Only the edge. The new N-1 is then stored with the import record. If you import recursively then those are also imported, up to N=0